src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs

```diff
@@ -388,6 +388,40 @@ public static bool TryGetActiveViewForTextBuffer(ITextBuffer textBuffer, out IWp
             return true;
         }
 
+        /// <summary>
+        /// Locates a tool window and request that it be shown.
+        /// </summary>
+        /// <param name="toolWindowGuid">A tool window identifier, typically one from <see cref="ToolWindowGuids80"/> such as the error list.</param>
+        /// <param name="activate">Indicates whether to activate (take keyboard focus) when showing the tool window.</param>
+        /// <returns>Returns a task that when completed indicates if the tool window was shown.</returns>
+        public static async System.Threading.Tasks.Task<bool> ShowToolWindowAsync(Guid toolWindowGuid, bool activate)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            IVsUIShell uiShell = Microsoft.VisualStudio.Shell.Package.GetGlobalService(typeof(SVsUIShell)) as IVsUIShell;
+            if (uiShell == null)
```

Comment:
Use null-conditional below to skip this check?

Reply:
This was lifted right from VS's codebase from their utility class.

Cleaned it up.

---
In reply to: [504106375](https://github.com/microsoft/sarif-visualstudio-extension/pull/223#discussion_r504106375) [](ancestors = 504106375)